src/sentry/receivers/onboarding.py

```diff
@@ -97,14 +97,15 @@ def record_new_project(project, user=None, user_id=None, **kwargs):
         ),
     )
 
-    success = OrganizationOnboardingTask.objects.record(
+    _, created = OrganizationOnboardingTask.objects.create_or_update(
```

Comment:
This should be fine for new project creation, but may not work as well for other onboarding signals that are higher frequency. Couldn't `create_or_update` be used within `OrganizationOnboardingTaskManager.record`?

Reply:
> This should be fine for new project creation, but may not work as well for other onboarding signals that are higher frequency

yes, that is why we are using the cache setter.

> Couldn't create_or_update be used within OrganizationOnboardingTaskManager.record?

We could update that function to use `create_or_update`, but I prefer not to touch it for now. Today I learned that we don't need to wrap everything with `with transaction.atomic` too, since there’s only one action being performed... the creation

Reply:
never mind - I applied your feedback here 🙂https://github.com/getsentry/sentry/pull/85929/commits/f21948f4cbc9d85adba8686b9e61533f6a871474